src/standalone/Vegetation/optimality_farquhar.jl

```diff
@@ -8,9 +8,12 @@ The required parameters for the optimality Farquhar photosynthesis model.
 Currently, only C3 photosynthesis is supported.
 $(DocStringExtensions.FIELDS)
 """
-Base.@kwdef struct OptimalityFarquharParameters{FT <: AbstractFloat}
+Base.@kwdef struct OptimalityFarquharParameters{
+    FT <: AbstractFloat,
+    MECH <: Union{FT, ClimaCore.Fields.Field},
+}
     "Photosynthesis mechanism: C3 only"
-    mechanism::C3
+    is_c3::MECH
```

Comment:
since we only allow for C3 right now with the Optimality model, should we @assert all(is_c3 .== 1) in the constructor?

Reply:
I followed the behavior of the previous constructor, which ignores any kwarg for is_c3 and sets is_c3 to FT(1). Should this be changed?

Reply:
I see that now, I think that is fine. I guess someone could set up the parameters using the default constructor with is_c3 = 0, but we note that this only supports C3 for now in the doc string.